src/main/java/com/alkemy/ong/exception/ExceptionHandler.java

```diff
@@ -0,0 +1,40 @@
+package com.alkemy.ong.exception;
+
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.FieldError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@ControllerAdvice
+public class ExceptionHandler extends ResponseEntityExceptionHandler {
+
+    public ExceptionHandler() {
+        super();
+    }
+
+    @Override
+    public ResponseEntity<Object> handleMethodArgumentNotValid(final MethodArgumentNotValidException ex, final HttpHeaders headers, final HttpStatus status, final WebRequest request) {
+        final Map<String, String> bodyOfResponse = new HashMap<>();
+        ex.getBindingResult().getAllErrors().forEach((error) -> {
+            String fieldName = ((FieldError) error).getField();
+            String errorMessage = error.getDefaultMessage();
+            bodyOfResponse.put(fieldName, errorMessage);
+        });
+        return handleExceptionInternal(ex, bodyOfResponse, headers, HttpStatus.BAD_REQUEST, request);
+    }
+
+    public ResponseEntity<ApiErrorDTO> handleOngExceptions(OngRequestException ex) {
+        ApiErrorDTO error =
+                ApiErrorDTO.builder()
+                        .code(ex.getCode())
+                        .message(ex.getMessage()).build();
+        return ResponseEntity.ok(error);
```

Comment:
Esto estaría mal, porque devolvemos un error pero con código 200 (indicando que está ok). A la exception le estaría faltando un campo de HttpStatus (para poner por ejemplo HttpStatus.NOT_FOUND) o un campo int en el que le indicamos el código (404 por ejemplo). Y eso es lo que usas para ponerle el código al ResponseEntity.

Reply:
Perfecto Santi! Ahí realicé las modificaciones. Lo del httpStatus lo pensé, pero está bien que todas las exceptions sean not found?